Libraries/Opc.Ua.Server/Subscription/MonitoredItem/MonitoredItem.cs

```diff
@@ -986,33 +964,21 @@ public virtual void QueueEvent(IFilterTarget instance, bool bypassFilter)
                 }
 
                 // can't do anything if queuing is disabled.
-                if (m_events == null)
+                if (m_eventQueueHandler == null)
                 {
                     return;
                 }
 
-                // check for duplicate instances being reported via multiple paths.
-                for (int ii = 0; ii < m_events.Count; ii++)
+                //Check for duplicates and discard
+                if (!m_eventQueueHandler.IsEventContainedInQueue(instance))
```

Comment:
I believe the not condition should be removed, or no events will be sent out.

This should also be tested with ConditionRefresh/ConditionRefresh2, which could be required to send out the event a second time.

Reply:
Thx for the catch. This was a To-Do for my unit Tests. u will fix it once I am back. conditions refresh is done in the subscription, not in the monitored item -> not affected by the Code Changes.